src/config.js

```diff
@@ -20,16 +21,30 @@ module.exports.WG_DEFAULT_DNS6 = typeof process.env.WG_DEFAULT_DNS6 === 'string'
     : '2606:4700:4700::1111';
 module.exports.WG_ALLOWED_IPS = process.env.WG_ALLOWED_IPS || '0.0.0.0/0, ::/0';
 
-module.exports.WG_POST_UP = process.env.WG_POST_UP || `
-iptables -t nat -A POSTROUTING -s ${module.exports.WG_DEFAULT_ADDRESS.replace('x', '0')}/24 -o eth0 -j MASQUERADE;
-iptables -A INPUT -p udp -m udp --dport 51820 -j ACCEPT;
-iptables -A FORWARD -i wg0 -j ACCEPT;
-iptables -A FORWARD -o wg0 -j ACCEPT;
-ip6tables -t nat -A POSTROUTING -s ${module.exports.WG_DEFAULT_ADDRESS6.replace('x', '0')}/120 -o eth0 -j MASQUERADE;
-ip6tables -A INPUT -p udp -m udp --dport 51820 -j ACCEPT;
-ip6tables -A FORWARD -i wg0 -j ACCEPT;
-ip6tables -A FORWARD -o wg0 -j ACCEPT;
-`.split('\n').join(' ');
+// Set WG_POST_UP to allow IPv6 NAT and forwarding only if the required kernel module is available
+const modules = childProcess.execSync('lsmod', {
+  shell: 'bash',
+})
+
+if (modules.includes("ip6table_nat")) {
+  module.exports.WG_POST_UP = process.env.WG_POST_UP || `
+  iptables -t nat -A POSTROUTING -s ${module.exports.WG_DEFAULT_ADDRESS.replace('x', '0')}/24 -o eth0 -j MASQUERADE;
+  iptables -A INPUT -p udp -m udp --dport 51820 -j ACCEPT;
+  iptables -A FORWARD -i wg0 -j ACCEPT;
+  iptables -A FORWARD -o wg0 -j ACCEPT;
+  ip6tables -t nat -A POSTROUTING -s ${module.exports.WG_DEFAULT_ADDRESS6.replace('x', '0')}/64 -o eth0 -j MASQUERADE;
+  ip6tables -A INPUT -p udp -m udp --dport 51820 -j ACCEPT;
+  ip6tables -A FORWARD -i wg0 -j ACCEPT;
+  ip6tables -A FORWARD -o wg0 -j ACCEPT;
+  `.split('\n').join(' ');
+} else {
+  module.exports.WG_POST_UP = process.env.WG_POST_UP || `
+  iptables -t nat -A POSTROUTING -s ${module.exports.WG_DEFAULT_ADDRESS.replace('x', '0')}/24 -o eth0 -j MASQUERADE;
+  iptables -A INPUT -p udp -m udp --dport 51820 -j ACCEPT;
+  iptables -A FORWARD -i wg0 -j ACCEPT;
+  iptables -A FORWARD -o wg0 -j ACCEPT;
+  `.split('\n').join(' ');
+}
```

Comment:
I think this block should be convert to something along the lines of this to avoid duplication:
```suggestion
module.exports.WG_POST_UP = process.env.WG_POST_UP
if (!!process.env.WG_POST_UP) {
  module.exports.WG_POST_UP = `
  iptables -t nat -A POSTROUTING -s ${module.exports.WG_DEFAULT_ADDRESS.replace('x', '0')}/24 -o eth0 -j MASQUERADE;
  iptables -A INPUT -p udp -m udp --dport 51820 -j ACCEPT;
  iptables -A FORWARD -i wg0 -j ACCEPT;
  iptables -A FORWARD -o wg0 -j ACCEPT;`
  
  if (modules.includes("ip6table_nat")) {
    module.exports.WG_POST_UP += `ip6tables -t nat -A POSTROUTING -s ${module.exports.WG_DEFAULT_ADDRESS6.replace('x', '0')}/64 -o eth0 -j MASQUERADE;
  ip6tables -A INPUT -p udp -m udp --dport 51820 -j ACCEPT;
  ip6tables -A FORWARD -i wg0 -j ACCEPT;
  ip6tables -A FORWARD -o wg0 -j ACCEPT;`
  }
  
  module.exports.WG_POST_UP = module.exports.WG_POST_UP.split('\n').join(' ');
}
```

Reply:
Yeah, that makes a lot of sense.  I kept it pretty basic (this is the first js project I've touched).